### DISCUSSION TITLE
Write support

### DISCUSSION THREAD

**Discussion by ZockiRR** [gotson/NightMonkeys#69] [Q&A]
Hi there,

I just wanted to ask if there are any plans to add encoding/writing support. Most interesting to me would be webp and avif.

Best regards

**Comment by gotson** [comment]
Hello,

i didn't need it myself so this was not added, but it would be possible. I am not familiar with the writing part of ImageIO though, and don't have much time to dedicate to that at the moment.

**Comment by ZockiRR** [comment]
Okay, thanks for the info. Unfortunately, I'm not familiar with the writing part of ImageIO nor the Foreign Function API. I didn't even get the project compiled. Maybe I can spend some more time and take a closer look, but no guarantees.

**Comment by gotson** [comment]
@ZockiRR i have added write support for HEIF/AVIF, it's available in the 1.1.0-SNAPSHOT. You can check the plugin's readme for details on how to configure the encoder.

Please share any feedback if you have.

**Reply by BuZZ-dEE** [reply]
@gotson how is it possible to write AVIF?

**Reply by gotson** [reply]
you need to set the `compressionType` to what you want
